test/distributed/org/apache/cassandra/distributed/test/AuthTest.java

```diff
@@ -41,15 +56,53 @@ public void authSetupIsCalledAfterStartup() throws IOException
-                    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
-
-                return StorageService.instance.authSetupCalled();
-            });
-            assertTrue(setupCalled);
+            IInvokableInstance instance = cluster.get(1);
+            await().pollDelay(1, SECONDS).pollInterval(1, SECONDS)
+                   .until(() -> instance.callOnInstance(() -> StorageService.instance.authSetupCalled()));
         }
     }
+
+    /**
+     * Sows that in some circumstances CassandraRoleManager will create the cassandra role twice
+     */
+    @Test
+    public void authSetupIsCalledTwice() throws IOException
+    {
+        try (Cluster cluster = builder().withNodes(1)
+                                        .withTokenSupplier(TokenSupplier.evenlyDistributedTokens(3))
+                                        .withNodeIdTopology(NetworkTopology.singleDcNetworkTopology(3, "dc0", "rack0"))
+                                        .withConfig(config -> config.with(NETWORK, GOSSIP).set("authenticator", "PasswordAuthenticator"))
+                                        .start())
+        {
+            IInvokableInstance instance = cluster.get(1);
+            await().pollDelay(1, SECONDS).pollInterval(1, SECONDS).atMost(12, SECONDS)
+                .until(() -> instance.callOnInstance(() -> CassandraRoleManager.hasExistingRoles()));
+
+            //get the time from the the first role setup
+            Long time1 = (Long)cluster.coordinator(1)
+                                      .execute("SELECT WRITETIME (salted_hash) from system_auth.roles where role = 'cassandra'",
+                                               ConsistencyLevel.ONE)[0][0];
+
+            IInstanceConfig config = cluster.newInstanceConfig();
+            // set boostrap to false to simulate a seed node
+            config.set("auto_bootstrap", false);
+            IInvokableInstance newInstance = cluster.bootstrap(config);
+            withProperty(BOOTSTRAP_SCHEMA_DELAY_MS.getKey(), Integer.toString(90 * 1000),
+                         () -> withProperty("cassandra.join_ring", false, () -> newInstance.startup(cluster)));
+            newInstance.nodetoolResult("join").asserts().success();
+
+            await().pollDelay(1, SECONDS).pollInterval(1, SECONDS)
+                     .until(() -> newInstance.callOnInstance(() -> CassandraRoleManager.hasExistingRoles()));
+
+            // get write titme frome second role setup
+            Long time2 = (Long)cluster.coordinator(1)
+                                      .execute("SELECT WRITETIME (salted_hash) from system_auth.roles where role = 'cassandra'",
+                                               ConsistencyLevel.ONE)[0][0];
+            // we don't do this here but if the user changed the Cassandra user password it will be (read) reapired if the second node has a later
+            // write timsestamp - check that this is not the case
+            assertTrue(time1 >= time2);
```

Comment:
why there is `>=` instead of just `=`? How can ever be `time1` bigger than `time2` anyway?